lib/Doctrine/DBAL/Exception/MissingSQLType.php

```diff
@@ -11,8 +11,6 @@ final class MissingSQLType extends SQLParserUtilsException
 {
     public static function new(string $typeName) : self
     {
-        return new self(
-            sprintf('Value for :%1$s not found in types array. Types array key should be "%1$s"', $typeName)
-        );
+        return new self(sprintf('Type named "%s" not found in types array.', $typeName));
```

Comment:
`SQLParserUtils::extractParam()` is really overloaded/generalized. What it's trying to say is that it's unable to find the type of the array parameter with the given name (`$paramName`, not `$typeName`). Something like this (add the missing articles):

> Type of array parameter %s is missing (not specified).

Reply:
It should be probably renamed to `MissingArrayParameterType` together with `MissingSQLParam` → `MissingArrayParameter`.

Reply:
Ok, I made another change. I think we're getting closer :)

Reply:
Definitely. But we may need a couple more passes :)